planner_cspace/test/src/test_grid_astar.cpp

```diff
@@ -91,6 +95,70 @@ TEST(GridAstar, ParallelSearch)
   }
 }
 
+TEST(GridAstar, SearchWithMultipleStarts)
+{
+  using Vec = CyclicVecInt<1, 1>;
+  GridAstar<1, 1> as(Vec(16));
+  as.setSearchTaskNum(1);
+
+  const auto cb_cost = [](
+      const Vec&, const Vec&, const std::vector<GridAstar<1, 1>::VecWithCost>&, const Vec&) -> float
+  {
+    return 1.0;
+  };
+  const auto cb_cost_estim = [](const Vec&, const Vec&) -> float
+  {
+    return 1e-6;
+  };
+  // create search table of graph edges (relative vector to the connected grid)
+  // 0: connected to 2-14
+  // 1: connected to 2-14
+  // 2-14: connected to 2-15
+  std::vector<Vec> search[16];
+  for (int i = 2; i < 15; ++i)
+  {
+    search[0].push_back(Vec(i));
+    search[1].push_back(Vec(i - 1));
+    for (int j = 2; j < 15; ++j)
+    {
+      if (i == j)
+        continue;
+      search[i].push_back(Vec(j - i));
+    }
+    search[i].push_back(Vec(15 - i));
+  }
+  const auto cb_search = [&search](
+      const Vec& p, const std::vector<GridAstar<1, 1>::VecWithCost>&, const Vec&) -> std::vector<Vec>&
+  {
+    return search[p[0]];
+  };
+  const auto cb_progress = [](const std::list<Vec>&)
+  {
+    return true;
+  };
+
+  for (int add_cost_to = 0; add_cost_to < 2; ++add_cost_to)
+  {
+    std::vector<GridAstar<1, 1>::VecWithCost> starts;
+    starts.emplace_back(Vec(0));
```

Comment:
"starts.emplace_back(0);" is better to avoid calling copy constructor.

Reply:
constructor used here is `VecWithCost(const Vec& v)`.

Reply:
https://github.com/at-wat/neonavigation/pull/351/files#diff-ee4bc0a1239b46a985e1df5a38d31823R61

Reply:
OK, I see